### PR TITLE
Adds ability to specify sanitizeOptions for prometheus metrics

### PR DESCRIPTION
**Pull request by mastermanu**
This PR adds the ability to set specific tally.SanitizeOptions via temporal configuration. If the configuration is not set, we rely on the default that is defined in code. Otherwise, we will generate a tally.SanitizeOptions by parsing the configuration file and converting it to a tally.SanitizeOptions.

Because configuration comes in via yaml, all relevant input fields are strings. The code converts the strings to runes and performs length check validations. Any invalid configuration will fail server startup.

### CODE REVIEW

File: common/metrics/config.go [hunk 1]
```diff
@@ -399,3 +443,64 @@ func configExcludeTags(cfg ClientConfig) map[string]map[string]struct{} {
 	}
 	return tagsToFilter
 }
+
+func (s SanitizeRange) toTally() (tally.SanitizeRange, error) {
+	startRangeRunes := []rune(s.StartRange)
+	if len(startRangeRunes) != 1 {
+		return tally.SanitizeRange{}, fmt.Errorf("start range '%+v' must be a single rune", startRangeRunes)
```
Comment (underrun) [inline common/metrics/config.go:450]: does this give enough feedback for someone using something that looks like a single character but isn't ala https://en.wikipedia.org/wiki/Combining_character - maybe it does but just want to bring it up.

Comment (mastermanu) [reply common/metrics/config.go:450]: i think the definition of a rune should make that clear as it maps to a Unicode Code Point

File: common/metrics/config.go [hunk 2]
```diff
@@ -399,3 +443,64 @@ func configExcludeTags(cfg ClientConfig) map[string]map[string]struct{} {
...
+	}
+
+	endRangeRunes := []rune(s.EndRange)
+	if len(endRangeRunes) != 1 {
+		return tally.SanitizeRange{}, fmt.Errorf("end range '%+v' must be a single rune", endRangeRunes)
+	}
+
+	return tally.SanitizeRange([2]rune{startRangeRunes[0], endRangeRunes[0]}), nil
```
Comment (yiminc) [inline common/metrics/config.go:458]: Do we need the 2 in [2]?

File: common/metrics/config.go [hunk 3]
```diff
@@ -399,3 +443,64 @@ func configExcludeTags(cfg ClientConfig) map[string]map[string]struct{} {
...
+
+	tallyKeyChars, err := s.KeyCharacters.toTally()
+	if err != nil {
+		return tally.SanitizeOptions{}, errors.New("invalid keyChars")
+	}
+
+	tallyValueChars, err := s.ValueCharacters.toTally()
+	if err != nil {
+		return tally.SanitizeOptions{}, errors.New("invalid valueChars")
+	}
```
Comment (yiminc) [inline common/metrics/config.go:493]: the error should wrap the detailed error.

File: common/metrics/config.go (hunk 2 above)
Comment (mastermanu) [reply common/metrics/config.go:458]: yes, that is the tally-specific type. they take in an array of runes, not a slice

File: common/metrics/config.go (hunk 3 above)
Comment (mastermanu) [reply common/metrics/config.go:493]: oops, good catch

**Review by yiminc — APPROVED** [review]

**Pull request merged**